src/frontend/org/voltdb/VoltTable.java

```
@@ -1596,6 +1596,34 @@ public String toJSONString() {
         return js.toString();
     }
 
+    /**
+     * Get a JSON /api/2.0/ representation of this table.
+     * @return A string containing a JSON representation of this table.
+     */
+    public JSONStringer toJSONStringerV2(JSONStringer js) {
+        try {
+            // array of row data
+            VoltTableRow row = cloneRow();
+            row.resetRowPosition();
+            js.array();
+            while (row.advanceRow()) {
+                js.object();
+                for (int i = 0; i < getColumnCount(); i++) {
+
+                    js.key(getColumnName(i));
+                    row.putJSONRep(i, js);
+                }
+                js.endObject();
+            }
+            js.endArray();
+        }
+        catch (JSONException e) {
```

I would just throw this exception from this method and let the caller handle it so there is one catch for this.